### DISCUSSION TITLE
"File name history" shows abbreviated paths that cannot be opened

### DISCUSSION THREAD

**Discussion by cweiske** [emacs-helm/helm#2674] [Q&A] [answered]
When opening helm's "find files" with `c-x c-f` I can open the recenly used files list with `c-c h` ("File name history").
In that list, the file paths are abbreviated with `…` characters. Selecting and opening them fails because the abbreviated path is used instead of the full path.

How can I fix this?

![2024-05-28 helm file name history](https://github.com/emacs-helm/helm/assets/59036/3e364abe-e8be-4c10-b779-660fa905e6d8)

This is with emacs 28.2 and helm 20240526.1107 on Debian 12.

**Comment by thierryvolpiatto** [comment]
Christian Weiske ***@***.***> writes:

> 1.  ( ) text/plain          (*) text/html           
>
> When opening helm's "find files" with c-x c-f I can open the recenly
> used files list with c-c h ("File name history").  In that list, the
> file paths are abbreviated with … characters. Selecting and opening
> them fails because the abbreviated path is used instead of the full
> path.

What make you think the "abbreviated and truncated candidate" is used?
We are using here a cons cell like (DISPLAY . REAL) where REAL is passed
to any action, I hardly see how it would fail, just tested right now
with "~/tmp/averylooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooonfile.txt"
which is truncated on display and it worked fine.

> How can I fix this?

I need more infos to help you on this (backtrace please preferably from
a minimal config i.e. no spacemacs or similar...).


-- 
Thierry

**Comment by thierryvolpiatto** [comment]
Note also that the files with red background in your screenshot are deleted files, you can toggle this view with `C-c d`.

**Comment by thierryvolpiatto** [comment, marked as answer]
This is a bug, thanks to catch this, the real candidate was indeed the truncated candidate, it is fixed now, please let me know if it works correctly after updating Helm (the commit will make its way to Melpa in a few hours).
Also I guess you have now in your `file-name-history` wrong candidates (the truncated ones), please cleanup `file-name-history` by editing this variable from M-x helm-apropos using the "Set variable" action.
Thanks.

**Reply by cweiske** [reply]
Thanks for the quick response and the fix!
It works now.
